File: python/0744_Find_Smallest_Letter_Greater_Than_Target.py
class Solution:
    def nextGreatestLetter(self, letters: List[str], target: str) -> str:
        left, right = 0, len(letters) - 1
        while left < right:
            mid = (left + right) // 2
            if letters[mid] <= target:
                left = mid + 1
            else:
                right = mid
        if letters[right] > target:
            return letters[right]
        return letters[0]

File: java/0744_Find_Smallest_Letter_Greater_Than_Target.java
class Solution {
    public char nextGreatestLetter(char[] letters, char target) {
        int left = 0, right = letters.length - 1;
        while (left < right) {
            int mid = left + (right - left) / 2;
            if (letters[mid] <= target) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }
        if (letters[right] > target) {
            return letters[right];
        }
        return letters[0];
    }
}